packages/tunnel-client/src/lib/tunnel.ts

```diff
@@ -0,0 +1,212 @@
+import { EventEmitter } from "events";
+import axios from "axios";
+import { Logger } from "loglevel";
+import { IncomingRequestEvent, LocalTunnelOptions, TunnelInfo } from "../types";
+import { TunnelCluster } from "./tunnel-cluster";
+
+const DEFAULT_HOST = "https://tunnels.meticulous.ai";
+
+interface CreateTunnelResponse {
+  id: string;
+  port: number;
+  url: string;
+  max_conn_count: number;
+  tunnel_passphrase: string;
+  basic_auth_user: string;
+  basic_auth_password: string;
+}
+
+interface CreateTunnelResponseError {
+  error: string;
+}
+
+export class Tunnel extends EventEmitter {
```

Comment:
Can you set the return types on this class' methods? With all the callbacks being passed around it's a bit confusing.

Reply:
Good shout, https://github.com/alwaysmeticulous/meticulous-sdk/pull/590/commits/32c4fc16363d479e113ec6561f9241fb2da2a73e does that